vscode/src/lspClient.ts

```diff
@@ -113,23 +113,17 @@ function startToitLsp(_: ExtensionContext,
 
   const result = new LanguageClient("toitLanguageServer", "Language Server", serverOptions, clientOptions);
   result.start();
-  let isReady = false;
-  let startFail = false;
-  setTimeout(() => {
-    // If the start failed, then VSCode already reported an error message.
-    // We could add an additional one to point the user to the trouble-shooting guide.
-    if (!startFail && !isReady) {
-      Window.showErrorMessage("The Language Server is not responding. Consult the documentation.");
-    }
-  }, 3000);
-  result.onReady().then(() => {
-    // At this point the language server responded to requests (`initialize`), and
-    //   is thus running.
-    isReady = true;
-  }, () => {
-    startFail = true;
+  return new Promise((resolve,reject) => {
+    setTimeout(() => {
+      // If the start failed, then VSCode already reported an error message.
+      // We could add an additional one to point the user to the trouble-shooting guide.
+      reject("The Language Server is not responding. Consult the documentation.");
```

Comment:
So you are relying on the fact that a Promise can't be completed more than once?
I would add a comment. I couldn't find good documentation for it (whether it's allowed or good practice), but it seems t work.

Reply:
It is fairly common way to make a timeout on a promise and we use it many places.